### PR TITLE
Messages

### LINKED ISSUE #21: Messages
Blocked by:
- [x] #16 

Parts:
- [ ] View for messages
- [ ] Adding new text messages
- [ ] Adding new media messages

### PR DESCRIPTION
**Pull request by mitas1**
To test this MR, create an project:
```sh
python manage.py shell
```

```python
>>> from buildinglog.models import Project
>>> p = Project(customer_id=1, project_manager_id=2, warranty=1, project_name="test")
>>> p.save()
```

then sign in as customer or employee:
```
username: customer
password: password

username: employee
password: password
```
This closes #21

### CODE REVIEW

**Comment by mskrip** [comment]
I assume you meant `project_name` instead of `name` in the project creation.

**Comment by mskrip** [comment]
Refreshing after sending the message creates a new message with the same content repeatedly.

**Comment by mitas1** [comment]
- `name` renamened to `project_name`
- in general, refreshing page after form submission causes form submission. Your browser should ask you if you really want to do that. I think this is expected. But if you have some idea how to resolve that you can provide your thoughts.

**Comment by mskrip** [comment]
If that is expected behavior then I have no complaints. Anyway, is there a way to restrain the type of file that is being uploaded to image? To not even have a choice to upload for example `.txt` file?

**Comment by mitas1** [comment]
We are using, `ImageField` and django validation for images. Uploading a `.txt` file should fails. If this is not your case, please attach that file and I am going to digging django source files :)

**Comment by mskrip** [comment]
No, upload fails. What I mean is I am offered to upload any file after clicking the upload button. Isn't there a quick way to forbid this?

**Comment by mitas1** [comment]
Yes, there is an option for that in HTML5. But unfortunately it is not implemented in materialize. Maybe we can create an issue for that in upstream, but it will be overhead. For now I created an issue #29 for that.

**Comment by mskrip** [comment]
LGTM

File: buildinglog/views/projects.py [hunk 1]
```diff
@@ -35,18 +35,42 @@ def new_item(request):
...
+                )
+        elif 'new_message' in request.POST and request.POST.get('message_text'):
+            message_form = MessageForm(
+                request.POST, request.FILES if len(request.FILES) != 0 else None)
+            if message_form.is_valid():
+                message_form = message_form.save(commit=False)
+                message_form.project = project
+                message_form.author = request.user
+                message_form.save()
+                notifications.append('Message was send')
```
Comment (mskrip) [inline buildinglog/views/projects.py]: Typo `sent`

**Comment by mskrip** [comment]
After uploading the image the page refreshes but the project parameters are missing. Also the image is not being uploaded now. It worked one now it is not working anymore.

File: buildinglog/views/projects.py [hunk 2]
```diff
@@ -35,18 +35,42 @@ def new_item(request):
...
     if request.method == 'POST':
-        if project_form.is_valid():
-            project_form.save()
+        if 'update' in request.POST:
+            if project_form.is_valid():
+                project_form.save()
+                notifications.append('Project was saved')
+            else:
+                notifications.append(
+                    'Some error occured during saving the project'
```
Comment (mskrip) [inline buildinglog/views/projects.py]: Something like `Some error occurred while saving the project` would be more appropriate.

**Comment by mskrip** [comment]
Also please add `media/` to `.gitignore`

**Comment by mitas1** [comment]
Fixed

**Pull request merged**